components/catalogue/EditProduct.jsx

```diff
@@ -0,0 +1,160 @@
+import {
+  Text,
+  SimpleGrid,
+  Button,
+  Center,
+  Box,
+  Image,
+  Stack,
+  Heading,
+  useColorModeValue,
+  Input,
+} from "@chakra-ui/react";
+import { EditIcon, ArrowBackIcon } from "@chakra-ui/icons";
+import { useState } from "react";
+
+export default function EditProduct(props) {
+  const product = props.product;
+
+  const [title, setTitle] = useState(product.title);
+  const [description, setDescription] = useState(product.description);
+  const [price, setPrice] = useState(product.price);
+  const [categories, setCategories] = useState(product.categories);
+  const [picture, setPicture] = useState(product.picture);
```

Comment:
Don't have an issue with this, but consider using an object as state or the `useReducer` hook